vllm/v1/core/scheduler.py

```diff
@@ -129,8 +133,11 @@ def schedule(self) -> "SchedulerOutput":
         req_index = 0
         while req_index < len(self.running) and token_budget > 0:
             request = self.running[req_index]
-            if request.request_id in self.scheduled_req_ids:
-                # This request has already been scheduled.
+            if (request.num_computed_tokens >= request.num_tokens
+                    and self.scheduled_req_ids.get(request.request_id, 0) > 0):
+                # We avoid re-scheduling the decoding requests because
+                # the number of new decoded output tokens is unknown due
+                # to speculative decoding or jump decoding.
```

Comment:
IIUC, this is not because of spec decoding. It's just because there's no token to schedule for this request.

Reply:
Orothogonally, I think it's worthwhile to add a comment on why we use `request.num_tokens` instead of `request.num_tokens_with_spec` here.

Reply:
Good point. Will update the comment.